src/arco_era5/source_data.py

```diff
@@ -372,12 +372,12 @@ def _read_nc_dataset(gpath_file):
         # and: https://confluence.ecmwf.int/display/CKB/ERA5%3A+data+documentation#ERA5:datadocumentation-Dataupdatefrequency  # pylint: disable=line-too-long
         # for further details.
 
-        all_dims_except_time = tuple(set(dataarray.dims) - {"time"})
+        all_dims_except_time = tuple(set(dataarray.dims) - {"time", "expver"})
         # Should have only trailing nans.
         a = dataarray.sel(expver=1).isnull().any(dim=all_dims_except_time)
```

Comment:
If I understand correctly the reason why this was not working is that `dataarray.sel(expver=1)` and `dataarray.sel(expver=5)`, 
even though they no longer have the "expver" dim, they both remember which expver coordinate they came from (they keep a scalar `expver` coordinate), and then they complain when running arithmetics between them because the scalar coordinate don't match. Then by adding "expver" to the dimensions to reduce the scalar coordinate diappears too. However this is unintuitive, because the propose code tries to reduce across a dim, which no longer exists.

I believe the right fix, rather than setting:
`all_dims_except_time = tuple(set(dataarray.dims) - {"time", "expver"})`

should be to use:
```
dataarray.sel(expver=1, drop=True)
dataarray.sel(expver=5, drop=True)
```

If you confirm this alternative fix works, this would be preferable.


I think the right fix for this should be:

Reply:
Talking more with Alvaro here: I think these errors only crop up if we download ERA5T, not ERA5. If we simply updated the data, or put limits into how recently we were updating the data, I believe these issues would go away.

Reply:
We've downloaded the data for 2023 on 21-22 September. Will that cause any trouble?
Tried below snipped giving an error.
```
all_dims_except_time = tuple(set(dataarray.dims) - {"time"})
a = dataarray.sel(expver=1, drop=True).isnull().any(dim=all_dims_except_time)
b = dataarray.sel(expver=5, drop=True).isnull().any(dim=all_dims_except_time)
disjoint_nans = bool(next(iter((a ^ b).all().data_vars.values())))
assert disjoint_nans, "The nans are not disjoint in expver=1 vs 5"
dataarray = dataarray.sel(expver=1).combine_first(dataarray.sel(expver=5))
```

Error
```
ValueError: 'expver' not found in array dimensions ('time', 'latitude', 'longitude')
```

Reply:
Could you indicate which line is throwing the error?

Perhaps the problem is that this file has an "expver" coordinate, but not an "expver" dim, which is something we have not found before, but can probably be solved by changing:

```
if "expver" in dataarray.coords:
   # previous code
```
to 
```
if "expver" in dataarray.dims:
  # previous code
elif "expver" in dataarray.coords:
  dataarray = dataarray.drop_vars('expver')
```

Reply:
> We've downloaded the data for 2023 on 21-22 September. Will that cause any trouble?

In short, yes I do think this is the root issue. Our goal with keeping data up-to-date is to have a ~1-2 month delay in freshness exactly for this reason. Unless we plan to re-download and update ERA5T data to ERA5 (I don't recommend this), then it would be best to change the interval at which we ingest data to avoid the case that this PR is trying to solve for.